include/fc/io/raw.hpp

```diff
@@ -193,10 +193,10 @@ namespace fc {
       uint64_t v = 0; char b = 0; uint8_t by = 0;
       do {
           s.get(b);
-          v |= uint32_t(uint8_t(b) & 0x7f) << by;
+          v |= uint64_t(uint8_t(b) & 0x7f) << by;
           by += 7;
-      } while( uint8_t(b) & 0x80 );
-      vi.value = static_cast<uint32_t>(v);
+      } while( (uint8_t(b) & 0x80) && by < 64 );
```

Comment:
Just a note: according to the discussion in https://github.com/steemit/steem/pull/2780, at a glance this change will break consensus, but our serialization code won't produce data to trigger it, so it's safe to sanitize inputs like this.